src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1943,14 +1949,8 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             case NI_Sve_TestFirstTrue:
             case NI_Sve_TestLastTrue:
             {
-                GenTree* op2 = retNode->AsHWIntrinsic()->Op(2);
-
                 // HWInstrinsic requires a mask for op2
-                if (!varTypeIsMask(op2))
-                {
-                    retNode->AsHWIntrinsic()->Op(2) =
-                        gtNewSimdCvtVectorToMaskNode(TYP_MASK, op2, simdBaseJitType, simdSize);
-                }
+                convertToMaskIfPossible(retNode->AsHWIntrinsic()->Op(2));
```

Comment:
I'm not quite sure this logic is "correct" (but recognize that it's not your PR making it that way, it's pre-existing).

Given this is import, we should always be getting `TYP_SIMD` off the evaluation stack and therefore should always need a `CvtVectorToMask` node.

The only thing that should be possible here is eliding `CvtVectorToMask` if `op2` was already `CvtMaskToVector` in which case we could consume `op2->AsHWIntrinsic()->Op(1)` directly instead.

Reply:
-- The reasoning for this is that hwintrinsics are imported as per the managed signature, which is always in terms of `TYP_SIMD`. So anything taking a `TYP_MASK` gets one by adding a `CvtVectorToMask` node and anything producing a `TYP_MASK` needs to wrap it with `CvtMaskToVector`

It's only as an optimization that we recognize `CvtVectorToMask(CvtMaskToVector(op))` and allow it to be consumed directly as `op`, so we shouldn't be encountering `TYP_MASK` nodes as direct operands here as part of `impHWIntrinsic`

Reply:
The logic isn't "incorrect", but it tells us that there are some cases in `impSpecialIntrinsic` that will put in the ConvertToMask - which is why we check to see if it's a mask or not at this point. Still, I think the check needs to exist because it also handles if you have one of the operands that pass an API that returns a mask.

Reply:
The reason its "incorrect" is because the `if (!varTypeIsMask(op))` should never be `false`

That is, we should always have a `TYP_SIMD` node here and therefore should always be needing to do `op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);`.

If `if (!varTypeIsMask(op))` is false, then we have a different bug that needs to be fixed instead.

Reply:
It doesn't need to be fixed in this PR, but we should log an issue to ensure that this logic is being properly handled and that we aren't creating invalid IR elsewhere

If another path is itself already handling the insertion, then it should be returning rather than hitting this path to ensure we can keep throughput higher.

Reply:
> we should always have a TYP_SIMD node here

Sure, but that is not what is happening right now, due to what I mentioned. - However, this is my assumption, I have yet to fully confirm it. Though, the logic in `implSpecialIntrinsic` needs changes as to not add `ConvertToMask` if we want `TYP_SIMD` to hold true for the `op` at that point.

Reply:
Added an issue: https://github.com/dotnet/runtime/issues/104836